Add PR template with the What's Changed entry convention (#4838)
icerpc/icerpc-csharp: .github/PULL_REQUEST_TEMPLATE.md

Reviewer: The name of this file seems incorrect. Looks like a mix-up between the file vs directory naming. See section 3:
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

---

For a single template (what we're doing) it should be a file named `pull_request_template.md`.
For multiple templates, it should be a directory named `PULL_REQUEST_TEMPLATE/`.

<!-- Describe the change and why it's needed. For a bug fix, describe the bug: when it occurs and what a user
observes. -->

## What's Changed entry

<!-- This section feeds the "What's Changed" section of the GitHub release notes; it's collected from merged PRs at

> release time. Keep it, in one of two forms.

Reviewer: I don't understand the current wording. What is there to "keep" here?
I feel like writing a simple declarative instructions is simpler:
```suggestion
release time. It should be in one of two forms:
```

> For a user-facing change, specify the area (usually an area label name, such as "Core", "Slic" or "Ice codec") and

Reviewer: We state that it should be in 1 of 2 forms, but the different forms are not obviously delineated.
Easily improved with:
```suggestion
1) For a user-facing change, specify the area (usually an area label name, such as "Core", "Slic" or "Ice codec") and
```

write one bullet per user-visible change:

Area: <area>

> - <entry>

Reviewer: Also, could be made cleaner with:
```suggestion
'''
Area: <area>

- <entry>
- 
'''
```

Write each entry from the user's perspective: the observable symptom or new behavior, in the documentation's
vocabulary — not the internal mechanism. A fix that only changes how invalid or malicious data received from a peer
is handled is not user-facing.

> 

Reviewer 2: I would just drop the example. I would say that if it can cause a crash and is and an important part of a patch release then it certainly warrants a changelog entry. 

```suggestion
vocabulary — not the internal mechanism.
```

> For a change with no user-facing impact, explain briefly why:

Reviewer: ```suggestion
2) For a change with no user-facing impact, explain briefly why:
```

> None — <short reason>.

Reviewer: ```suggestion
'''
None — <short reason>.
'''
```
